tstdc/src/mutex.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Mutex functionality for an SGX enclave
```

Comment:
For my own understanding, is this `Mutex` meant to be used by multiple SGX processor threads? Or is used by multiple non-SGX threads accessing enclave code? (I may be asking the wrong question, but I'm curious as to how threading works in SGX and how this struct will be used.)

Reply:
Thanks for keeping me honest here.

I added more detail to the Mutex struct as that shows up in the generated documentation but the module documentation doesn't.
I did update to `Mutex functionality for use inside of an SGX enclave` here to try to clarify, but I think the `Mutex` docs will contain most of the docs.

Also trying to think of where to better document `root ECALL`.